source/MaterialXShaderGen/HwShaderGenerator.h

```diff
@@ -0,0 +1,48 @@
+#ifndef MATERIALX_HWSHADERGENERATOR_H
+#define MATERIALX_HWSHADERGENERATOR_H
+
+#include <MaterialXShaderGen/ShaderGenerator.h>
+
+namespace MaterialX
+{
+
+using HwShaderGeneratorPtr = shared_ptr<class HwShader>;
+
+/// Base class for shader generators targeting HW rendering.
+class HwShaderGenerator : public ShaderGenerator
+{
```

Comment:
New base class with light binding features. Calling these methods to bind light shaders is done on application side, where there is knowledge of which light types to support. This is done before any call to `generate()`, since in order to generate a surface shader we need light shader knowledge first.

Reply:
Hi Niklas, Can you explain more about how this works. If the bind is by light type then doesn't that mean you can only bind 1 light of each type?

Reply:
Sure. The "binding" here is for _light shaders_ not _light sources_. So it's for telling the generator which light shader types should be supported in generated surface shader, to avoid emitting code for _all_ possible light types, if just a subset of light types are supported by the target system. And it's also assigning light type ids to these light shaders, so light type id's doesn't have to be universally defined and can be set by the application.

Binding _light sources_ is another thing and done by binding data to the u_lightData[] array, and setting u_numActiveLightSources. Here the application sets the light type id for each entry in the array, and any number of light sources can be created for each light type.